lms/djangoapps/commerce/api/v1/serializers.py

```diff
@@ -17,6 +18,14 @@ def get_identity(self, data):
         except AttributeError:
             return None
 
+    def validate(self, data):
+        data = super(CourseModeSerializer, self).validate(data)
+        if 'professional' in data['mode_slug'] and data['expiration_datetime'] is not None:
```

Comment:
With my recent changes, there's really only one mode that should have an expiration datetime: "verified".  None of the other course modes have an upgrade deadline (although "professional" may have a verification deadline).